### DISCUSSION TITLE
ZBM with Encrypted Dataset I don't want mounted / OnFailure=emergency.target ??

### DISCUSSION THREAD

**Discussion by reefland** [zbm-dev/zfsbootmenu#413] [Questions] [answered]
I've created a dataset such as `k3s06/backups`  within that I have remote system place their encrypted datasets as backups.  This part is working fine.

Upon reboot ZBM, asks for `k3s06/ROOT` password as expected and I can enter this via DropBear too.

However, after entering this password ZBM exists out and the system starts to boot and then is prompting me for password of the `k3s06/backups` dataset(s).  At this point it is after DropBear but before SSHD is available, so this is only shown on the console.

If I just do CTRL-C or press ENTER either of these result in:
```
[FAILED] Failed to start Load ZFS key for rpool/backups/apps
[DEPEND] Dependency failed for /backups/apps
````
The system essentially halts. Will not progress any further. The system is unbootable as each datasets use a different long hex password.

On the Traditional OpenZFS on Root method I simply set `canmount=noauto` on these remote datasets and that is enough to not be prompted and system is allowed to boot.  I even tried  changing `keylocation` from `prompt` to a filepath that does not exist and it still prompts for a password.

Using the ZBM recovery console, I had to destroy all datasets under `k3s06/backups` to get the system bootable again.  

I also tried creating the dataset under `k3s06/ROOT/backups` and then after remote replication completed, I set `canmount=noauto` on its `k3s06/ROOT/backups/apps` and rebooted.  It correctly did not prompt me for a dataset password, but still left the system unbootable with message:
```
[FAILED] Failed to start Import keys for all datasets"
````
And would not progress beyond that. 

Again use ZBM recovery console to delete the datasets and the system was bootable again.

Lastly I modified the `/etc/systemd/system/zfs-load-key.service` and changed `OnFailure=emergency.target` to just `OnFailure=`

Upon rebooting, enter the normal ZBM encryption password and then there was just a time delay of some sort - minute or so (no prompt for password, no error messages) but just wait and the system booted up as expected. No errors within `dmesg` output.

So what is the expected behavior of `OnFailure=emergency.target` when using ZBM / Dracut ??  Should that have done something such as drop me back to ZBM recovery console instead of a system hang / halt ?

**Comment by ahesford** [comment, marked as answer]
This is your distribution's initramfs. ZFSBootMenu has no control over the boot process after it launches your boot environment.
